File: FinalProject/app/src/main/java/com/example/finalproject/service/spell/Element.java
package com.example.finalproject.service.spell;

public class Element extends Component{
    private int element, baseDamage;

    public Element(String name, int element, int baseDamage, boolean available) {
        super(name , available);
        this.element = element;
        this.baseDamage = baseDamage;
    }

    public Element(Element element){
        super(element.getName(), element.isAvailable());
        this.element=element.element;
        baseDamage =element.baseDamage;
    }

    public int getElement() {
        return element;
    }

    public int getBaseDamage() {
        return baseDamage;
    }
}

File: FinalProject/app/src/main/java/com/example/finalproject/service/spell/Element.kt
package com.example.finalproject.service.spell

class Element(name:String, private val element:Int, val baseDamage:Double, available:Boolean):Component(name, available) {
    constructor(element: Element) : this(element.name, element.element, element.baseDamage, element.available)

    fun getElement()=element
}